packages/core/src/types/IImage.ts

```diff
@@ -10,6 +10,8 @@ import { ImageQualityStatus } from '../enums';
 interface IImage {
   /** Image Id */
   imageId: string;
+  /** referenced imageId if this image is derived from */
+  referencedImageId?: string;
```

Comment:
This should be an array and should reference all the overlays that go into the image.

Reply:
One image will only have always one reference, this is not about segmentation